Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/local/source/LocalStatisticsProvider.kt

```diff
@@ -0,0 +1,128 @@
+package de.rki.coronawarnapp.statistics.local.source
+
+import de.rki.coronawarnapp.statistics.StatisticsData
+import de.rki.coronawarnapp.statistics.local.FederalStateToPackageId
+import de.rki.coronawarnapp.statistics.local.storage.LocalStatisticsConfigStorage
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.device.ForegroundState
+import de.rki.coronawarnapp.util.flow.HotDataFlow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import org.joda.time.Duration
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class LocalStatisticsProvider @Inject constructor(
+    @AppScope private val scope: CoroutineScope,
+    private val server: LocalStatisticsServer,
+    private val localStatisticsCacheFactory: LocalStatisticsCache.LocalStatisticsCacheFactory,
+    private val localStatisticsConfigStorage: LocalStatisticsConfigStorage,
+    private val localStatisticsParser: LocalStatisticsParser,
+    foregroundState: ForegroundState,
+    dispatcherProvider: DispatcherProvider
+) {
+
+    private val localStatisticsData = HotDataFlow(
+        loggingTag = TAG,
+        scope = scope,
+        coroutineContext = dispatcherProvider.IO,
+        sharingBehavior = SharingStarted.WhileSubscribed(
+            stopTimeoutMillis = Duration.standardSeconds(5).millis,
+            replayExpirationMillis = 0
+        )
+    ) {
```

Comment:
I think this is missing a flow subscription that monitors this and on change, writes all changes back to storage?